osu.Game/Screens/Select/BeatmapInfoWedge.cs

```diff
@@ -153,5 +187,53 @@ public void UpdateBeatmap(WorkingBeatmap beatmap)
                 Add(d);
             });
         }
+
+        private string getBPMRange(Beatmap beatmap)

```

Comment:
We might want to make this part of `Beatmap`, but I am unsure. Just to keep direct handling of control points more encapsulated. Split in `getMinBPM`, `getMaxBPM`, `getMostFrequentBPM`, and keep string assembly in here.